app/src/main/java/org/jellyfin/androidtv/auth/AuthenticationRepository.kt

```diff
@@ -142,7 +145,30 @@ class AuthenticationRepositoryImpl(
 			// Access token found, proceed with sign in
 			account?.accessToken != null -> when {
 				// Update session
-				setActiveSession(user, server) -> emit(AuthenticatedState)
+				setActiveSession(user, server) -> {
+					// Update stored user
+					try {
+						val response = UserApi(userApiClient).getCurrentUser()
+						val userInfo = response.content
+						val currentUser = authenticationStore.getUser(server.id, user.id)
+
+						val updatedUser = currentUser?.copy(
+							name = userInfo.name!!,
+							requirePassword = userInfo.hasPassword,
+							imageTag = userInfo.primaryImageTag
+						) ?: AuthenticationStoreUser(
+							name = userInfo.name!!,
+							requirePassword = userInfo.hasPassword,
+							imageTag = userInfo.primaryImageTag
+						)
+						authenticationStore.putUser(server.id, user.id, updatedUser)
+						accountManagerHelper.putAccount(AccountManagerAccount(user.id, server.id, updatedUser.name, account.accessToken))
+					} catch(err: ApiClientException) {
+						Timber.e(err, "Unable to get current user data")
```

Comment:
I think we want to emit `RequireSignInState` when an apiclient exception occurs, since apparantly the authentication is bad